Objects/Converters/ConverterRevit/ConverterRevitShared/Partial Classes/UpdateParameter.cs

```diff
@@ -3,44 +3,50 @@
 using Objects.Geometry;
 using System.Collections.Generic;
 using System.Linq;
+using Speckle.Core.Helpers;
+using Speckle.Core.Models;
 
 
 namespace Objects.Converter.Revit
 {
   public partial class ConverterRevit
   {
-    public void UpdateParameter(ParameterUpdater paramUpdater)
+    public ApplicationObject UpdateParameter(ParameterUpdater paramUpdater)
     {
       //the below does not work because ApplicationIds are stored within a stream
       //try to get element using ApplicationId
       //this will only work if the element has been successfully been received in Revit before
       //var element = GetExistingElementByApplicationId(paramUpdater.revitId);
-
+      
+      var appObj = new ApplicationObject(paramUpdater.id, paramUpdater.speckle_type) { applicationId = paramUpdater.applicationId };
       Element element = null;
 
       //try to get element using ElementId
-      int intId;
-
-      if (int.TryParse(paramUpdater.elementId, out intId))
+      if (int.TryParse(paramUpdater.elementId, out int intId))
       {
         var elemId = new ElementId(intId);
         element = Doc.GetElement(elemId);
       }
 
       //try to get element using UniqueId
-      if (element == null)
+      element ??= Doc.GetElement(paramUpdater.elementId);
+
+      if (element != null)
       {
-        element = Doc.GetElement(paramUpdater.elementId);
+        SetInstanceParameters(element, paramUpdater);
+        appObj.Update(status: ApplicationObject.State.Updated, logItem: $"Successfully updated instance parameters for element {paramUpdater.elementId}");
       }
-
-      if (element == null)
+      else
       {
-        Report.LogConversionError(new System.Exception($"Could not find element to update: Element Id = {paramUpdater.elementId}"));
-        return;
+        Report.LogConversionError(
+          new System.Exception($"Could not find element to update: Element Id = {paramUpdater.elementId}"));
```

Comment:
I mean, this is kind of interesting....  we call this method passing it an exception. Also, do we have a better exception, if not, can we please? :)

Reply:
Agreed! Any particular suggestion?

If we need to create a brand new exception for this, where should it live?

Reply:
I don't know what we currently have, also, where does what we currently have live? Somewhere common is the answer there though

Reply:
I've removed this line altogether as it turned out to be unnecessary, flagging this conversation as resolved 😄